lib/mongo/crypt/auto_encrypter.rb

```diff
@@ -53,6 +53,17 @@ module AutoEncrypter
       def setup_encrypter(options = {})
         opts = set_default_options(options.dup)
 
+        unless opts[:key_vault_client]
+          # If no key vault client is passed in, create one by duplicating the
+          # Mongo::Client used for encryption. Update options so that key vault
+          # client does not perform auto-encryption/decryption, and keep a reference
+          # to it so it is destroyed later.
+          @key_vault_client = self.dup
```

Comment:
Is this dup immune to the issue described in https://jira.mongodb.org/browse/RUBY-1506?jql=project%20%3D%20RUBY%20AND%20status%20%3D%20Open%20AND%20text%20~%20dup ?

Reply:
Does `#with` work to change `auto_encryption_options` to nil?

Reply:
RE: your first comment -- you are very right. I didn't know about that. Thank you so much for catching it.

Reply:
I'm adding a test that makes sure key_vault_client is a functional client and fixing this now 👍